spacy/cli/_util.py

```diff
@@ -215,30 +220,71 @@ def validate_project_version(config: Dict[str, Any]) -> None:
         msg.fail(err, exits=1)
 
 
+def validate_max_parallel_processes(config: Dict[str, Any]) -> None:
+    """If the project defines a maximum number of parallel processes, check that the
+    value is within the permitted range.
+
+    config (Dict[str, Any]): The loaded config.
+    """
+    max_parallel_processes = config.get("max_parallel_processes", None)
+    if max_parallel_processes is not None and max_parallel_processes < 2:
+        err = (
+            f"The {PROJECT_FILE} specifies a value for max_parallel_processes ({max_parallel_processes}) "
+            f"that is less than 2."
+        )
+        msg.fail(err, exits=1)
+
+
+def verify_workflow_step(workflow_name: str, commands: List[str], step: str) -> None:
+    if step not in commands:
+        msg.fail(
+            f"Unknown command specified in workflow '{workflow_name}': {step}",
+            f"Workflows can only refer to commands defined in the 'commands' "
+            f"section of the {PROJECT_FILE}.",
+            exits=1,
+        )
+
+
 def validate_project_commands(config: Dict[str, Any]) -> None:
     """Check that project commands and workflows are valid, don't contain
     duplicates, don't clash  and only refer to commands that exist.
 
     config (Dict[str, Any]): The loaded config.
     """
-    command_names = [cmd["name"] for cmd in config.get("commands", [])]
+
+    commands = [cmd["name"] for cmd in config.get("commands", [])]
     workflows = config.get("workflows", {})
-    duplicates = set([cmd for cmd in command_names if command_names.count(cmd) > 1])
+    duplicates = set([cmd for cmd in commands if commands.count(cmd) > 1])
     if duplicates:
         err = f"Duplicate commands defined in {PROJECT_FILE}: {', '.join(duplicates)}"
         msg.fail(err, exits=1)
-    for workflow_name, workflow_steps in workflows.items():
-        if workflow_name in command_names:
+    for workflow_name, workflow_items in workflows.items():
+        if workflow_name in commands:
             err = f"Can't use workflow name '{workflow_name}': name already exists as a command"
             msg.fail(err, exits=1)
-        for step in workflow_steps:
-            if step not in command_names:
-                msg.fail(
-                    f"Unknown command specified in workflow '{workflow_name}': {step}",
-                    f"Workflows can only refer to commands defined in the 'commands' "
-                    f"section of the {PROJECT_FILE}.",
-                    exits=1,
-                )
+        for workflow_item in workflow_items:
+            if isinstance(workflow_item, str):
+                verify_workflow_step(workflow_name, commands, workflow_item)
+            else:
+                assert isinstance(workflow_item, dict)
+                assert len(workflow_item) == 1
+                steps_list = workflow_item["parallel"]
+                assert isinstance(steps_list[0], str)
+                steps = cast(List[str], steps_list)
```

Comment:
You're right that this would be better, but this also raises the general point you made above about asserts: perhaps we should be throwing a `ValueError` here instead?

Reply:
I looked it up since I wasn't sure about how strict we are about this. Our code conventions mention that
> As a general rule of thumb, all error messages raised within the core library should be added to `Errors`.

So I'd say let's convert all raw asserts into `ValueError`s.

Reply:
In fact the project file has already been validated against the schema, so these `assert`s are not really necessary in the first place: I'll get rid of them (as well as implementing your suggestion).